src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Windows.cs

```diff
@@ -1087,5 +1087,57 @@ internal static unsafe SocketError DuplicateSocket(SafeSocketHandle handle, int
                 return result == 0 ? SocketError.Success : GetLastSocketError();
             }
         }
+
+        internal static unsafe bool HasNonBlockingConnectCompleted(SafeSocketHandle handle, out bool success)
+        {
+            bool refAdded = false;
+            try
+            {
+                handle.DangerousAddRef(ref refAdded);
+
+                IntPtr rawHandle = handle.DangerousGetHandle();
+                IntPtr* writefds = stackalloc IntPtr[2] { (IntPtr)1, rawHandle };
+                IntPtr* exceptfds = stackalloc IntPtr[2] { (IntPtr)1, rawHandle };
+                Interop.Winsock.TimeValue IOwait = default;
```

Comment:
Nit: I know this is copy of existing code, but consider renaming `IOWait` -> `timeout`.